cpp/src/parquet/metadata.h

```diff
@@ -510,24 +510,23 @@ class PARQUET_EXPORT FileMetaDataBuilder {
  public:
   // API convenience to get a MetaData reader
   static std::unique_ptr<FileMetaDataBuilder> Make(
-      const SchemaDescriptor* schema, std::shared_ptr<WriterProperties> props,
-      std::shared_ptr<const KeyValueMetadata> key_value_metadata = NULLPTR);
+      const SchemaDescriptor* schema, std::shared_ptr<WriterProperties> props);
```

Comment:
I know it makes the code more complex, but I think for now we should create a new overload that drops the `key_value_metadata` parameter and add `PARQUET_DEPRECATED` marker to the existing constructor.

Reply:
Agree

Reply:
Sounds reasonable. Fixed.

The only caveat is that the old overload simply ignores the input `key_value_metadata` which may confuses users.

Reply:
Ignoring the parameter seems like an even worse breaking change than removing it. Could we simply make this backwards compatible? Either `key_value_metadata` in `Finish` in override, or it should merge with the one provided in the constructor.